image/store.go

```diff
@@ -67,21 +67,21 @@ func NewImageStore(fs StoreBackend, lss LayerGetReleaser) (Store, error) {
 
 func (is *store) restore() error {
 	err := is.fs.Walk(func(dgst digest.Digest) error {
-		img, err := is.Get(IDFromDigest(dgst))
+		img, err := is.Get(ID(dgst))
 		if err != nil {
-			logrus.Errorf("invalid image %v, %v", dgst, err)
+			logrus.WithField("digest", dgst).WithError(err).Error("invalid image")
 			return nil
 		}
 		var l layer.Layer
 		if chainID := img.RootFS.ChainID(); chainID != "" {
```

Comment:
here we can shadow previous log in this scope with a new `log` with chainID and os:
```suggestion
		if chainID := img.RootFS.ChainID(); chainID != "" {
			log := logrus.WithField("chainID", chainID).WithField("os", img.OperatingSystem())
```